test/test_cuda_nvml_based_avail.py

```diff
@@ -0,0 +1,69 @@
+# Owner(s): ["module: cuda"]
+
+import os
+import sys
+import multiprocessing
+import torch
+import unittest
+from unittest.mock import patch
+
+# NOTE: Each of the tests in this module need to be run in a brand new process to ensure CUDA is uninitialized
+# prior to test initiation.
+with patch.dict(os.environ, {"PYTORCH_NVML_BASED_CUDA_CHECK": "1"}):
+    # Before executing the desired tests, we need to disable CUDA initialization and fork_handler additions that would
+    # otherwise be triggered by the `torch.testing._internal.common_utils` module import
+    from torch.testing._internal.common_utils import (parametrize, instantiate_parametrized_tests, run_tests, TestCase,
+                                                      IS_WINDOWS)
+    # NOTE: Because `remove_device_and_dtype_suffixes` initializes CUDA context (triggered via the import of
+    # `torch.testing._internal.common_device_type` which imports `torch.testing._internal.common_cuda`) we need
```

Comment:
cc @mruberry, @pmeier this cuda context initialization seems unneeded (no action necessary on this PR @speediedan)

Reply:
That's true. This happens because `common_device_type.py` imports from `common_cuda.py`, which unconditionally initializes the CUDA context. We could probably refactor that import . Would you file an issue?